packages/toast/src/Toast.tsx

```diff
@@ -0,0 +1,301 @@
+
+  dismissButton: css`
+    position: absolute;
+    top: ${spacing[2]}px;
+    right: ${spacing[2]}px;
+    transition: color 0.15s ease-in-out;
+  `,
+};
+
+const ToastVariants = {
+  success: 'success',
+  progress: 'progress',
+} as const;
+
+type ToastVariants = typeof ToastVariants[keyof typeof ToastVariants];
+
+const variantStyles: Record<
+  ToastVariants,
+  Partial<Record<StyledElements, string>>
+> = {
+  [ToastVariants.success]: {
+    toast: css`
+      background-color: ${uiColors.green.light3};
+    `,
+
+    icon: css`
+      color: ${uiColors.green.base};
+    `,
+
+    contentWrapper: css`
+      border-radius: 4px;
+      border: 1px solid ${uiColors.green.light2};
+    `,
+
+    body: css`
+      color: ${uiColors.green.dark2};
+    `,
+
+    dismissButton: css`
+      color: ${uiColors.green.base};
+
+      &:hover {
+        color: ${uiColors.green.dark2};
+
+        &:before {
+          background-color: ${uiColors.green.light2};
+        }
+      }
+    `,
+  },
+
+  [ToastVariants.progress]: {
+    toast: css`
+      background-color: ${uiColors.white};
+      padding-bottom: 6px;
+    `,
+
+    icon: css`
+      color: ${uiColors.gray.dark2};
+    `,
+
+    contentWrapper: css`
+      border-radius: 4px 4px 0 0;
+      border: 1px solid ${uiColors.blue.light2};
+      border-bottom: 0;
+    `,
+
+    body: css`
+      color: ${uiColors.gray.dark2};
+    `,
+  },
+};
+
+const RTGStates = {
+  Entering: 'entering',
+  Entered: 'entered',
+  Exiting: 'exiting',
+  Exited: 'exited',
+} as const;
+
+type RTGStates = typeof RTGStates[keyof typeof RTGStates];
+
+const toastTransitionStateStyles: Partial<Record<RTGStates, string>> = {
+  [RTGStates.Entered]: css`
+    transform: translate3d(0, 0, 0) scale(1);
+    opacity: 1;
+  `,
+};
+
+interface ToastProps extends Omit<React.ComponentProps<'div'>, 'title'> {
+  /**
+   * Optional text shown in bold above the body text.
+   */
+  title?: React.ReactNode;
+
+  /**
+   * Required main text for the Toast.
+   */
+  body: React.ReactNode;
+
+  /**
+   * Optional className passed to the wrapping <div /> for the toast.
+   */
+  className?: string;
+
+  /**
+   * Required style variant to render the Toast as.
+   */
+  variant: ToastVariants;
+
+  /**
+   * Optional number between 0 and 1 that sets the progress bar's progress. Note that the progress bar is only rendered when the Toast variant is set to `'progress'`.
+   *
+   * **Default:** `1`
+   */
+  progress?: number;
+
+  /**
+   * Optional boolean that renders the Toast and makes it visible when true.
+   *
+   * **Default:** `false`
+   */
+  open?: boolean;
+
+  /**
+   * Optional click event handler that, when set, renders a close button that receives the passed handler.
+   */
+  close?: React.MouseEventHandler;
+}
+
+function Toast({
+  title,
+  body,
+  className,
+  variant,
+  progress = 1,
+  open = false,
+  close,
+  ...rest
+}: ToastProps) {
+  const nodeRef = useRef(null);
+  const dismissible = !!close;
```

Comment:
```suggestion
  const dismissible = typeof close === 'function';
```

For 'correctness' sake?

Reply:
I had a feeling you'd say something about coercing this to a boolean like this 😄  Happy to make the change – I do think your suggestion is correct here.